app/controllers/training_data/dashboard_controller.rb

```diff
@@ -1,19 +1,13 @@
 class TrainingData::DashboardController < ApplicationController
-  before_action :guard_trainer!, except: :become_trainer
+  before_action :guard_trainer!
 
   def index
     @training_data_dashboard_params = params.permit(:status, :order, :criteria, :page, :track_slug)
     @statuses = %w[needs_tagging needs_checking needs_checking_admin]
   end
 
-  def become_trainer
-    User::BecomeTrainer.(current_user) unless current_user.trainer?
-
-    redirect_to training_data_root_path
-  end
-
   def guard_trainer!
-    return true if current_user&.trainer?
+    return true if current_user.trainer?
```

Comment:
I think this still needs the `&` as logged in users get here by default before being redirected.

Reply:
I thought that the default was to require an authenticated user. If I go to that page without being logged-in, I get redirected to the login page.